src/prefect/utilities/logging.py

```diff
@@ -305,14 +305,20 @@ class RedirectToLog:
     def __init__(self, logger: logging.Logger = None) -> None:
         self.stdout_logger = logger or get_logger("stdout")
 
-    def write(self, s: str) -> None:
+    def write(self, s: Union[str, bytes]) -> None:
         """
         Write message from stdout to a prefect logger.
         Note: blank newlines will not be logged.
 
         Args:
             s (str): the message from stdout to be logged
         """
+        if isinstance(s, bytes):
```

Comment:
I'm a bit confused by this. When we capture `stdout`, all we do is replace `sys.stdout` in the task context with a new file-like object. Normally `sys.stdout` is a `io.TextIOWrapper`, which takes in `str` (not `bytes`) as input. No file type should ever get a mix of `str` or `bytes` under proper operation, I'm confused why `click` is sending it bytes. Perhaps our mock file class here is missing an interface method that click relies on to infer whether to send bytes or str?

Can you provide a reproducible example so we can investigate deeper? While the above might fix it, I don't think it's the best fix.